shop/src/main/java/ro/msg/learning/shop/config/LocationStrategyConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package ro.msg.learning.shop.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.strategies.LocationStrategy;
+import ro.msg.learning.shop.strategies.MostAbundantStrategy;
+import ro.msg.learning.shop.strategies.SingleLocationStrategy;
+
+@Configuration
+public class LocationStrategyConfiguration {
+
+    @Value("${location.strategy}")
+    private String locationStrategy;
+
+    @Autowired
+    private SingleLocationStrategy singleLocationStrategy;
+
+    @Autowired
+    private MostAbundantStrategy mostAbundantStrategy;
+
+    @Bean
+    public LocationStrategy locationStrategy() {
+        if ("custom".equals(locationStrategy)) {
```

Comment:
use a constant when possible. Also "custom" does not reflec the fact that its most abundant. "abundant" would tho

Reply:
for you it makes 100% sense now the naming, however, in a few days/weeks it wont. and for someone new it would create confusion. Even if it works, not respecting (or loosely respecting) good practices such as naming can cause the following effects in the future:
confusion -> more time to develop -> possible flaws in features -> hard to spot bugs -> low coding standrs -> more confusion and so on.
This app is just practice,obviously, but your standards and principles which you apply in practice and in real apps will define who you are in terms of software craftmanship. And being a good craftsman will leade to better projects for you and your team. People will follow your example and bit by bit software will become better around you. But this can only happen if you develop such principles and to do that you have to apply them everywhere, regardless if its just practice, just for fun, or a real app with real steakholders.